### PR TITLE
No duplicate matches

### PR DESCRIPTION
**Pull request by Dbz**
I noticed that my HS Tracker would create many duplicate matches, and I would have to manually delete them. This will stop the creation of duplicate matches.

I COULD NOT GET HEARHTSTATS TO BUILD ON MY SYSTEM. Therefore, I was not actually able to test this, nor run the tests. I realize this is a big inconvenience for someone else to test, but I would greatly appreciate it. If this does not work, please tell me and I will update the code.

This pull request contains four things:

1) An `==` method for matches
2) A `no_duplicate_matches` validation for matches
3) A spec for the validation
4) Enforcing single/double quote styleguide rules as requested on the readme to gemfile, match.rb, and match_spec.rb

### CODE REVIEW

**Comment by trigun0x2** [comment]
For some reason I did not get a notification for this PR! Reviewing!

File: app/models/match.rb [hunk 1]
```diff
@@ -390,6 +392,19 @@ def update_user_stats_constructed
...
   end
 
+  def ==(match)
+    self.oppname == match.oppname && self.duration == match.duration
+  end
+
+  ### VALIDATION METHODS:
+
+  def no_duplicate_matches
+    user = User.find(self.user_id)
```
Comment (trigun0x2) [inline app/models/match.rb]: Every match created now has to look up the user again. I wonder if we can skip this in some way?

Does the dupes come in a row or in separate times?

**Comment by Dbz** [comment]
@trigun0x2 , thank you for reviewing my pull request. I appreciate it. I was only interested in the last user match, so it was easy to remove the user query.

I'm not sure what you mean by coming in a row or separate times, but I can describe what happens:

I win/lose/tie a game on HS and my tracker sends the match information to hearthstats. I look under the recent games in constructed, and there are multiple copies of the match (sometimes even an entire page+ of the last match I played). This seriously messes up the statistics because all of a sudden I have 20 more losses, or 20 more wins. Then I need to delete each match (and wait for the page to reload).

**Comment by trigun0x2** [comment]
That's really weird why that would happen.. But I've merged the PR! :+1:

**Comment by trigun0x2** [comment]
@Dbz can you go ahead and test the patch? It's deploted

**Comment by Dbz** [comment]
@trigun0x2, will do. I'm currently traveling over the next 48 hours, but I will post an update as soon as I play some matches.

**Pull request merged**